Mathlib/Combinatorics/SimpleGraph/StronglyRegular.lean

```diff
@@ -206,4 +210,23 @@ theorem IsSRGWith.param_eq (h : G.IsSRGWith n k ℓ μ) (hn : 0 < n) :
       ← Set.toFinset_card]
     congr!
 
+/-- Let `A` and `C` be the adjacency matrices of a strongly regular graph with parameters `n k ℓ μ`
+and its complement respectively and `I` be the identity matrix,
+then `A ^ 2 = k • I + ℓ • A + μ • C`. `C` is equivalent to the expression `J - I - A`
+more often found in the literature, where `J` is the all-ones matrix. -/
+theorem IsSRGWith.matrix_eq [Semiring α] (h : G.IsSRGWith n k ℓ μ) :
+    G.adjMatrix α ^ 2 = k • (1 : Matrix V V α) + ℓ • G.adjMatrix α + μ • Gᶜ.adjMatrix α := by
+  ext v w
+  simp only [adjMatrix_pow_apply_eq_card_walk, Set.coe_setOf, Matrix.add_apply, Matrix.smul_apply,
+    adjMatrix_apply, compl_adj]
+  letI : Fintype {p : G.Walk v w // p.length = 2} := G.fintypeSetWalkLength v w 2
```

Comment:
This should be a global instance, immediately after where `fintypeSetWalkLength` is defined.